File: pset2/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <math.h>
#include <ctype.h>


int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{

    //prompt to get User Input
    string text = get_string("Text: ");

    //Declare the var which include num of elements in text.
    int num_letters = count_letters(text);
    int num_words = count_words(text);
    int num_sentences = count_sentences(text);

    //calcualte index
    float L = num_letters / (float)num_words * 100;
    float S = num_sentences / (float)num_words * 100;

    int index = round(0.0588 * L - 0.296 * S - 15.8);


    if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }

}


int count_letters(string text)
{
    int letters = 0;

    for (int i = 0 ; i < strlen(text) ; i++)
    {
        if (isalpha(text[i]))
        {
            letters++;
        }
        else
        {

        }
    }
    return letters;
}

int count_words(string text)
{
    //At the end of the word in the sentence does not have space, so that set initial value as '1'
    int words = 1;

    for (int i = 0 ; i < strlen(text) ; i++)
    {
        if (text[i] == 32)
        {
            words ++;
        }
        else
        {

        }
    }
    return words;
}


int count_sentences(string text)
{
    
    int sentences = 0;

    for (int i = 0 ; i < strlen(text) ; i++)
    {
        if (text[i] == 46 || text[i] == 33 || text[i] == 63)
        {
            sentences ++;
        }
        else
        {

        }
    }
    return sentences;
}

File: pset6/readability/readability.py
import cs50

def main():
 
    #prompt to get User Input as string type
    text = cs50.get_string("Text: ")
    
    #Declare the var which include number of elements in text.
    num_letters = count_letters(text)
    num_words = count_words(text)
    num_sentences = count_sentences(text)
    
    #Calculate index
    L = num_letters / float(num_words) * 100
    S = num_sentences / float(num_words) * 100
    
    index = round(0.0588 * L - 0.296 * S - 15.8)
    print(index)
    
    if (index >= 16):
        print("Grade 16+")
    elif (index < 1):
        print("Before Grade 1")
    else: 
        print("Grade ",index)
    
    
    
    
    
    

#define function that count number of letters, words and sentences
def count_letters(text):
    letters = 0
    
    for i in range(0, len(text)):
        if (str.isalpha(text[i])):
            letters += 1
    #print(letters)
    return letters


def count_words(text):
    
    #at the end of the word in the sentence does not have space, then set initial value as '1'.
    words = 1
    
    for i in range(0, len(text)):
        if (ord(text[i]) == 32):
            words += 1
   #print(words)
    return words


def count_sentences(text):
    
    sentences = 0;
    
    for i in range(0, len(text)):
        # 46: '.', 33: '!', 63: '?'  in ascii code
        if (ord(text[i]) == 46 or ord(text[i]) == 33 or ord(text[i]) == 63):
            sentences += 1
    #print(sentences)
    return sentences



if __name__ == "__main__":
    main()
